src/platforms/common/swdptap.c

```diff
@@ -182,13 +199,16 @@ static void swdptap_seq_out(const uint32_t tms_states, const size_t clock_cycles
 
 static void swdptap_seq_out_parity(const uint32_t tms_states, const size_t clock_cycles)
 {
-	int parity = __builtin_popcount(tms_states);
+	const int parity = __builtin_popcount(tms_states);
```

Comment:
As this is only used in the one place and is bitwise masked, perhaps lets make this `const uint8_t parity` and put the bit mask on the end, lifting it from the `gpio_set_val()` call below? Failing that, we should make it `uint32_t` so we don't do a signed+bitwise oops below.

Reply:
Zooming out: do we actually care about 2..32 number of bits, or just the SWD parity per IHI0031 B4.1.6? The same 4698a26ba introduced the two builtin calls, which was a better idea than spinning the parity manually. And I see an opportunity to eliminate either `__popcountsi2()` or `__paritysi2()`, currently both are used, only once. They are 32+40 bytes, (36+52 depending on platform Thumb2 availability; popcount is bigger). GCC sees we don't care in seq_out and substitutes parity in place of popcount (because of the LSBit masking), but doesn't see in seq_in and leaves popcount alone.

Reply:
Thread resolved by merging (and rebasing over) PR1700.